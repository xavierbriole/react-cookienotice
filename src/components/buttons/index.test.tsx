import React from 'react'
import { render } from '@testing-library/react'

import Buttons from './index'

describe('Buttons', () => {
  beforeEach(() => {
    jest.resetAllMocks()
  })

  describe('with open in new tab false', () => {
    it('should render', () => {
      const { container } = render(
        <Buttons
          acceptButtonLabel='acceptButtonLabel'
          handleAcceptButtonClick={() => {}}
          readMoreButtonLabel='readMoreButtonLabel'
          readMoreButtonLink='readMoreButtonLink'
          readMoreButtonOpenInNewTab={false}
        />,
      )

      expect(container.firstChild).toMatchSnapshot()
    })
  })

  describe('with open in new tab true', () => {
    it('should render', () => {
      const { container } = render(
        <Buttons
          acceptButtonLabel='acceptButtonLabel'
          handleAcceptButtonClick={() => {}}
          readMoreButtonLabel='readMoreButtonLabel'
          readMoreButtonLink='readMoreButtonLink'
          readMoreButtonOpenInNewTab={true}
        />,
      )

      expect(container.firstChild).toMatchSnapshot()
    })
  })
})
