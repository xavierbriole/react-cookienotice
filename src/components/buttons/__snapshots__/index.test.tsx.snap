// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buttons with open in new tab false should render 1`] = `
<div
  class="react-cookienotice-buttons-wrapper"
>
  <button
    class="react-cookienotice-accept-button"
  >
    <span>
      acceptButtonLabel
    </span>
  </button>
  <a
    class="react-cookienotice-read-more-button"
    href="readMoreButtonLink"
    target="_self"
  >
    readMoreButtonLabel
  </a>
</div>
`;

exports[`Buttons with open in new tab true should render 1`] = `
<div
  class="react-cookienotice-buttons-wrapper"
>
  <button
    class="react-cookienotice-accept-button"
  >
    <span>
      acceptButtonLabel
    </span>
  </button>
  <a
    class="react-cookienotice-read-more-button"
    href="readMoreButtonLink"
    rel="noreferrer"
    target="_blank"
  >
    readMoreButtonLabel
  </a>
</div>
`;
